plugins/amazonq/chat/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonq/webview/BrowserConnector.kt

```diff
@@ -167,32 +171,59 @@ class BrowserConnector(
 
-                val partialResultToken = chatCommunicationManager.addPartialChatMessage(requestFromUi.params.tabId)
                 val chatParams = ChatParams(
                     requestFromUi.params.tabId,
                     chatPrompt,
                     textDocumentIdentifier,
                     cursorState
                 )
 
+                val tabId = requestFromUi.params.tabId
+                val partialResultToken = chatCommunicationManager.addPartialChatMessage(tabId)
+
                 var encryptionManager: JwtEncryptionManager? = null
                 val result = AmazonQLspService.executeIfRunning(project) { server ->
                     encryptionManager = this.encryptionManager
                     encryptionManager?.encrypt(chatParams)?.let { EncryptedChatParams(it, partialResultToken) }?.let { server.sendChatPrompt(it) }
                 } ?: (CompletableFuture.failedFuture(IllegalStateException("LSP Server not running")))
+                showResult(result, partialResultToken, tabId, encryptionManager, browser)
+            }
+            CHAT_QUICK_ACTION -> {
+                val requestFromUi = serializer.deserializeChatMessages(node, QuickChatActionRequest::class.java)
+                val tabId = requestFromUi.params.tabId
+                val quickActionParams = requestFromUi.params
+                val partialResultToken = chatCommunicationManager.addPartialChatMessage(tabId)
+                var encryptionManager: JwtEncryptionManager? = null
+                val result = AmazonQLspService.executeIfRunning(project) { server ->
+                    encryptionManager = this.encryptionManager
+                    encryptionManager?.encrypt(quickActionParams)?.let {
+                        EncryptedQuickActionChatParams(it, partialResultToken)
+                    }?.let {
+                        server.sendQuickAction(it)
+                    }
+                } ?: (CompletableFuture.failedFuture(IllegalStateException("LSP Server not running")))
 
-                result.whenComplete {
-                        value, error ->
-                    chatCommunicationManager.removePartialChatMessage(partialResultToken)
-                    val messageToChat = ChatCommunicationManager.convertToJsonToSendToChat(
-                        node.command,
-                        requestFromUi.params.tabId,
-                        encryptionManager?.decrypt(value).orEmpty(),
-                        isPartialResult = false
-                    )
-                    browser.postChat(messageToChat)
-                }
+                showResult(result, partialResultToken, tabId, encryptionManager, browser)
             }
         }
     }
+
+    private fun showResult(
+        result: CompletableFuture<String>,
+        partialResultToken: String,
+        tabId: String,
+        encryptionManager: JwtEncryptionManager?,
+        browser: Browser,
+    ) {
+        result.whenComplete { value, error ->
```

Comment:
need to handle error

Reply:
Being handled here: https://github.com/aws/aws-toolkit-jetbrains/pull/5571/files#diff-81c603e8d34d45cba14b619394dec7a76ff291c3ec9fb994ec5c611f80f6c30aR202